app/src/main/java/com/example/tasky/feature_login/domain/model/AuthenticationViewState.kt

```diff
@@ -1,7 +1,9 @@
 package com.example.tasky.feature_login.domain.model
 
+import com.example.tasky.common.data.util.DataError
+
 open class AuthenticationViewState {
     object Success : AuthenticationViewState()
-    data class Failure(val message: String) : AuthenticationViewState()
+    data class Failure(val dataError: DataError) : AuthenticationViewState()
     object Loading : AuthenticationViewState()
```

Comment:
Also if you have a class with a fixed set of options, better make that a sealed class/interface. Then you get proper smart casting and when-exhaustiveness :)